utils/mail.go

```diff
@@ -104,36 +104,56 @@ func TestConnection(config *model.Config) {
 }
 
 func SendMailUsingConfig(to, subject, htmlBody string, config *model.Config) *model.AppError {
+	fromMail := mail.Address{Name: config.EmailSettings.FeedbackName, Address: config.EmailSettings.FeedbackEmail}
+	return sendMail(to, to, fromMail, subject, htmlBody, nil, nil, config)
+}
+
+// allows for sending an email with attachments and differing MIME/SMTP recipients
+func SendMailUsingConfigAdvanced(mimeTo, smtpTo string, from mail.Address, subject, htmlBody string, attachments []*model.FileInfo, mimeHeaders map[string]string, config *model.Config) *model.AppError {
+	return sendMail(mimeTo, smtpTo, from, subject, htmlBody, attachments, mimeHeaders, config)
+}
+
+func sendMail(mimeTo, smtpTo string, from mail.Address, subject, htmlBody string, attachments []*model.FileInfo, mimeHeaders map[string]string, config *model.Config) *model.AppError {
```

Comment:
GlobalRelay emails are quite a bit different than the normal emails that the system already sends. To support the various odds and ends that the system requires, I needed to overload the existing `SendEmailUsingConfig()` method to allow the user to specify a bunch of details.

Unfortunately, Golang doesn't support method overloading (I really miss Java some days), so this is the best I could do. I'm not 100% jazzed about the number of parameters in the advanced method signature, but it is called advanced, so...